server/src/main/java/io/druid/server/AsyncQueryForwardingServlet.java

```diff
@@ -313,6 +317,13 @@ protected void sendProxyRequest(
     // will log that on the remote node.
     clientRequest.setAttribute(AuthConfig.DRUID_AUTHORIZATION_CHECKED, true);
 
+    authenticatorMapper.getAuthenticatorChain()
```

Comment:
This would potentially "overdecorate" the request if `decorateProxyRequest` is called for every Authenticator, suppose there are two authentication schemes that need request decoration but have mutually exclusive header fields, or some other conflict of that nature.

I would change AuthenticationResult to store the name of the Authenticator that created the AuthenticationResult, and read that name here to lookup the single Authenticator instance that authenticated the request, only that one should decorate the request

Reply:
@jon-wei am not sure what you mean by mutually exclusive headers? can you give an example for instance here am using cookies that can be redecorated by appending stuff not sure am getting this point?

> I would change AuthenticationResult to store the name of the Authenticator that created the AuthenticationResult, and read that name here to lookup the single Authenticator instance that authenticated the request, only that one should decorate the request

From this i read there is only one `AuthenticationResult` thus this implies that we only have one single authenticator even if the user has a chain of authenticators? is that correct ?
thanks

Reply:
That "stop at first successful authentication" behavior is enforced by `AuthenticationWrappingFilter` which wraps all of the Authenticator filters